drivers/led/led_pwm.c

```diff
@@ -115,7 +115,7 @@ static int led_pwm_pm_action(const struct device *dev,
 		LOG_DBG("PWM %p running pm action %" PRIu32, led->dev, action);
 
 		err = pm_device_action_run(led->dev, action);
-		if (err && (err != -EALREADY)) {
+		if (err && (err != -EALREADY) && (err != -ENOSYS)) {
 			LOG_ERR("Cannot switch PWM %p power state", led->dev);
```

Comment:
Where does the flood come from ?

I understand we may log an error each time we attempt to change the power state because the PM callback is missing. But eventually it is an interesting information and it is not supposed to flood the logs.

In addition I think it should be useful to display the `err` value in the log.

Reply:
Yes, the flood comes from the attempts to change the power state, which can happen many times per second.

I do agree this could be use information (cough cough LOG_INF) but I certainly do not think this is an error as it's perfectly acceptable for the underlying PWM device to not implement any sort of PM functionality. 

Would you prefer changing to LOG_INF (maybe just for the ENOSYS case?)? Or any other suggested solution? Open to ideas here as I just need a way to stop getting spammed when using a PWM device without a PM callback.

Thanks you for reviewing my PR!

Reply:
> Yes, the flood comes from the attempts to change the power state, which can happen many times per second.

It is this part I'd like to clarify: is that OK to switch between the off, suspending, suspended and active power states "many" times per second ?

I'd like to ensure that masking the error (or information) message is not masking another issue.

Reply:
The frequency in which the system goes from PM_STATE_ACTIVE to a state that suspends devices (PM_STATE_SUSPEND_TO_IDLE, PM_STATE_STANDBY, PM_STATE_SUSPEND_TO_RAM, etc) depends on both system activity and the latency in which is takes to enter such states.

Consider a very simple system that reads some sensor data at 10 Hz and sleeps in between. With a 100 ms sleep period there is more than enough time to enter a deep sleep state such as PM_STATE_STANDBY - a minimum of 2 ms is all that is needed for the pm policy for the SoC that I am using to enter PM_STATE_STANDBY. In this system we would get spammed with the "error" at 20 Hz (once when entering and once when exiting suspend) when there is in fact no error.

Reply:
Thank for taking the time to explain the issue to me !

Don't you think he flood is likely to happen with other scenario ? For example a PM action callback will return `-ENOTSUP` for a unsupported action. Should we also exclude the `-ENOTSUP` error code from the log ? Besides if a callback runs into an error, it is likely to happen again at the next attempt...

So I'm wondering if it is correct for the `led_pwm` driver to log errors at all here. What do you think ?

Reply:
Yes I do agree there are other scenarios such as the `-ENOTSUP` case you've mentioned. Instead of excluding each error code, I'm beginning to think we should just change `LOG_ERR` to `LOG_DBG` (`LOG_INF` would also be fine in my eyes) given this would all be useful information when debugging power related discrepancies. How does this sound?

Reply:
It sounds good. My preference goes to `LOG_DBG`, but I am OK with `LOG_INF` too.

Reply:
Excellent, I have just update the commit to change `LOG_ERR` to `LOG_DBG`

Reply:
@simonguinot @dennisgr102 Maybe it will be worth printing the return code if it is possible that a different errno will be thrown? As it's a DBG log it would be useful to include more information about the error.

Reply:
Yes @bbilas. Please @dennisgr102, display the error code as well.

Reply:
Thanks for the feedback, just updated my commit!